pkg/network/infraconfigurators/bridge.go

```diff
@@ -50,21 +51,34 @@ func (b *BridgePodNetworkConfigurator) DiscoverPodNetworkInterface(podIfaceName
 	}
 	b.podNicLink = link
 
+	b.ipamEnabled = false
+
 	addrList, err := b.handler.AddrList(b.podNicLink, netlink.FAMILY_V4)
 	if err != nil {
 		log.Log.Reason(err).Errorf("failed to get an ip address for %s", podIfaceName)
 		return err
 	}
-	if len(addrList) == 0 {
-		b.ipamEnabled = false
-	} else {
-		b.podIfaceIP = addrList[0]
+	if len(addrList) > 0 {
+		b.podIfaceIP = &addrList[0]
 		b.ipamEnabled = true
 		if err := b.learnInterfaceRoutes(); err != nil {
 			return err
 		}
 	}
 
+	addrV6List, err := b.handler.AddrList(b.podNicLink, netlink.FAMILY_V6)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get an ipv6 address for %s", podIfaceName)
+		return err
+	}
+	for _, addr := range addrV6List {
+		if addr.IP.IsGlobalUnicast() {
+			b.podIfaceIPv6 = &addr
+			b.ipamEnabled = true
+			break
+		}
+	}
```

Comment:
can't we export this into a func parametrized by IP family ? You could even receive a variadic func parameter for extra operations (i.e. invoke `b.learnInterfaceRoutes()` for ipv4).

Reply:
Yeah, makes sense to do this. Trying to make it more generic for handling e.g. route learning as well didn't really end up looking more elegant to me, but that's a style question I guess.